go-controller/pkg/ovn/controller/networkconnect/topology.go

```diff
@@ -0,0 +1,913 @@
+				continue
+			}
+
+			// Calculate nexthop (second IP of the P2P subnet on network router side)
+			p2pSubnets, err := calculateP2PSubnets(subnets, nodeID)
+			if err != nil {
+				continue
+			}
+			_, networkPortIPs := getP2PIPs(p2pSubnets)
+			nexthops := util.IPNetsToIPs(networkPortIPs)
+
+			// Create route for this node's subnets
+			ops, err = c.createStaticRoutesOps(ops, networkOwner, connectRouterName, nodeSubnets, nexthops, cncName, nodeID)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	if netInfo.TopologyType() == ovntypes.Layer2Topology {
+		// For Layer2, create a single route to the network's subnets
+		_, networkPortIPs := getP2PIPs(subnets)
+		nexthops := util.IPNetsToIPs(networkPortIPs)
+
+		var podSubnetIPNets []*net.IPNet
+		for _, entry := range podSubnets {
+			podSubnetIPNets = append(podSubnetIPNets, entry.CIDR)
+		}
+
+		ops, err = c.createStaticRoutesOps(ops, networkOwner, connectRouterName, podSubnetIPNets, nexthops, cncName, 0)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ops, nil
+}
+
+// createStaticRoutesOps returns ops to create logical router static routes.
+func (c *Controller) createStaticRoutesOps(ops []ovsdb.Operation, networkOwner, routerName string, dstSubnets []*net.IPNet, nexthops []net.IP, cncName string, nodeID int) ([]ovsdb.Operation, error) {
+	for _, dstSubnet := range dstSubnets {
+		// Find matching nexthop (same IP family)
+		ipFamily := "v4"
+		var nexthop string
+		for _, nh := range nexthops {
+			isIPv4Subnet := utilnet.IsIPv4(dstSubnet.IP)
+			isIPv4Nexthop := utilnet.IsIPv4(nh)
+			if isIPv4Subnet == isIPv4Nexthop {
+				nexthop = nh.String()
+				if !isIPv4Subnet {
+					ipFamily = "v6"
+				}
+				break
+			}
+		}
+		if nexthop == "" {
+			continue
+		}
+
+		dbIndexes := libovsdbops.NewDbObjectIDs(libovsdbops.LogicalRouterStaticRouteClusterNetworkConnect, controllerName,
+			map[libovsdbops.ExternalIDKey]string{
+				libovsdbops.NetworkKey:    networkOwner,
+				libovsdbops.NodeIDKey:     strconv.Itoa(nodeID),
+				libovsdbops.ObjectNameKey: cncName, // CNC name
+				libovsdbops.IPFamilyKey:   ipFamily,
+			})
+		route := &nbdb.LogicalRouterStaticRoute{
+			IPPrefix:    dstSubnet.String(),
+			Nexthop:     nexthop,
+			ExternalIDs: dbIndexes.GetExternalIDs(),
+		}
+
+		var err error
+		// Don't limit fields to update - when node subnets change, IPPrefix and Nexthop need to be updated too
+		ops, err = libovsdbops.CreateOrUpdateLogicalRouterStaticRoutesWithPredicateOps(c.nbClient, ops, routerName, route,
+			libovsdbops.GetPredicate[*nbdb.LogicalRouterStaticRoute](dbIndexes, nil))
+		if err != nil {
+			return nil, fmt.Errorf("failed to create static route ops on %s: %v", routerName, err)
+		}
+
+		klog.V(5).Infof("Created/updated static route ops on %s: %s via %s", routerName, dstSubnet.String(), nexthop)
+	}
+
+	return ops, nil
+}
+
+// findNetworkByID finds a network by its ID.
+func (c *Controller) findNetworkByID(networkID int) (util.NetInfo, error) {
```

Comment:
maybe we can move this to the networkManager interface, I don't like locking outside of the networkManager

Reply:
maybe we could also add some more efficient way to get network by ID there if needed..